server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -276,25 +301,27 @@ public static Query fromPrimitive(DataType<?> type, String column, Object value)
      * </pre>
      **/
     private static Query refEqObject(Function eq,
-                                     String fqn,
+                                     ColumnIdent columnIdent,
                                      ObjectType type,
                                      Map<String, Object> value,
-                                     LuceneQueryBuilder.Context context) {
+                                     Context context) {
         BooleanQuery.Builder boolBuilder = new BooleanQuery.Builder();
         int preFilters = 0;
         for (Map.Entry<String, Object> entry : value.entrySet()) {
             String key = entry.getKey();
             DataType<?> innerType = type.innerType(key);
-            if (innerType == null) {
+            if (innerType == UndefinedType.INSTANCE) {
```

Comment:
Why did you have to change this? Is there a test that covers this case?

Reply:
It is because `type.innerType` is a `non-null` method, calling

```
innerTypes.getOrDefault(key, UndefinedType.INSTANCE)
```

Reply:
I will added a test for it, Thank you

Reply:
thx!